### ISSUE TITLE
segfault when running cluster_fast with msaout

### ISSUE THREAD

**Issue by avilella** [torognes/vsearch#76]
Hi @torognes ,

I am testing `--cluster_fast` with `--msaout` and for some of my clusters, I get segfaults like the one below (input `test.fasta` is below):

Running with this command:

```
vsearch-1.0.10/bin/vsearch-1.0.10-linux-x86_64 --minseqlength 1 --msaout test.fasta.msa --cluster_fast /tmp/test.fasta --id 0.97
```

I get:

```
vsearch v1.0.10_linux_x86_64, 126.0GB RAM, 48 cores
https://github.com/torognes/vsearch

Reading file /tmp/test.fasta 100%  
2100 nt in 50 seqs, min 42, max 42, avg 42
Indexing sequences 100% 
Masking 100%
Sorting by length 100%
Counting unique k-mers 100% 
Clustering 100%  
Writing clusters 100% 
Clusters: 1 Size min 50, max 50, avg 50.0
Singletons: 0, 0.0% of seqs, 0.0% of clusters
Multiple alignments 100% 
Segmentation fault (core dumped)
```

```
>GC:42021428:.:.:ATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTA:86.6875:PASS:C1=16;C2=1387;EF=86.6875;GC=0.214286;DU=0.6;VI=-2.2
ATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTA
>GC:42021428:.:.:TTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAA:150.222:PASS:C1=9;C2=1352;EF=150.222;GC=0.214286;DU=0.5;VI=-1.9
TTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAA
>GC:42021428:.:.:TTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAA:114.75:PASS:C1=12;C2=1377;EF=114.75;GC=0.214286;DU=0.575;VI=-2.2
TTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAA
>GC:42023809:.:.:CATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTT:74.2632:PASS:C1=19;C2=1411;EF=74.2632;GC=0.238095;DU=0.5;VI=-2.2
CATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTTT
>GC:42023809:.:.:TGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAG:90.3333:PASS:C1=15;C2=1355;EF=90.3333;GC=0.238095;DU=0.475;VI=-1.9
TGTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAG
>GC:42026190:.:.:AAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTAT:75.2632:PASS:C1=19;C2=1430;EF=75.2632;GC=0.261905;DU=0.3;VI=-2.2
AAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTAT
>GC:42026190:.:.:AGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATT:75:PASS:C1=19;C2=1425;EF=75;GC=0.261905;DU=0.375;VI=-2.2
AGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATT
>GC:42026190:.:.:GCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTT:47.5667:PASS:C1=30;C2=1427;EF=47.5667;GC=0.261905;DU=0.425;VI=-2.2
GCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTATTT
>GC:42026190:.:.:GTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGC:223.5:PASS:C1=6;C2=1341;EF=223.5;GC=0.261905;DU=0.475;VI=-1.9
GTTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGC
>GC:42026190:.:.:TAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTA:96:PASS:C1=15;C2=1440;EF=96;GC=0.261905;DU=0.3;VI=-2.2
TAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACTA
>GC:42026190:.:.:TTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCG:63.8571:PASS:C1=21;C2=1341;EF=63.8571;GC=0.261905;DU=0.45;VI=-1.9
TTTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCG
>GC:42028571:.:.:AAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTAT:68.4286:PASS:C1=21;C2=1437;EF=68.4286;GC=0.285714;DU=0.275;VI=-7.3
AAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTAT
>GC:42028571:.:.:AAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATA:63.0435:PASS:C1=23;C2=1450;EF=63.0435;GC=0.285714;DU=0.3;VI=-7.3
AAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATA
>GC:42028571:.:.:AATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATC:152.4:PASS:C1=10;C2=1524;EF=152.4;GC=0.285714;DU=0.325;VI=-7.3
AATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATC
>GC:42028571:.:.:AATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAA:40.2778:PASS:C1=36;C2=1450;EF=40.2778;GC=0.285714;DU=0.325;VI=-7.3
AATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAA
>GC:42028571:.:.:ATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCT:135.727:PASS:C1=11;C2=1493;EF=135.727;GC=0.285714;DU=0.3;VI=-7.3
ATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCT
>GC:42028571:.:.:CAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATAT:103.333:PASS:C1=15;C2=1550;EF=103.333;GC=0.285714;DU=0.3;VI=-7.6
CAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATAT
>GC:42028571:.:.:CTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACT:110.308:PASS:C1=13;C2=1434;EF=110.308;GC=0.285714;DU=0.325;VI=-2.7
CTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAACT
>GC:42028571:.:.:TAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTA:69.5238:PASS:C1=21;C2=1460;EF=69.5238;GC=0.285714;DU=0.275;VI=-7.3
TAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTA
>GC:42028571:.:.:TTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGAT:82:PASS:C1=17;C2=1394;EF=82;GC=0.285714;DU=0.375;VI=-0.8
TTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGAT
>GC:42028571:.:.:TTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGC:70.7368:PASS:C1=19;C2=1344;EF=70.7368;GC=0.285714;DU=0.375;VI=-1.5
TTCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGC
>GC:42030952:.:.:ACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAAT:88.9444:PASS:C1=18;C2=1601;EF=88.9444;GC=0.309524;DU=0.225;VI=-7.3
ACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAAT
>GC:42030952:.:.:AGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGT:73.1667:PASS:C1=18;C2=1317;EF=73.1667;GC=0.309524;DU=0.375;VI=-4.3
AGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGT
>GC:42030952:.:.:ATACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGA:139.917:PASS:C1=12;C2=1679;EF=139.917;GC=0.309524;DU=0.25;VI=-7.3
ATACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGA
>GC:42030952:.:.:ATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAAC:63.4348:PASS:C1=23;C2=1459;EF=63.4348;GC=0.309524;DU=0.325;VI=-7.3
ATCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAAC
>GC:42030952:.:.:CCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATA:198.625:PASS:C1=8;C2=1589;EF=198.625;GC=0.309524;DU=0.275;VI=-7.3
CCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAATA
>GC:42030952:.:.:CCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAA:129.545:PASS:C1=11;C2=1425;EF=129.545;GC=0.309524;DU=0.3;VI=-7.3
CCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAA
>GC:42030952:.:.:CCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAAC:83.2941:PASS:C1=17;C2=1416;EF=83.2941;GC=0.309524;DU=0.3;VI=-6
CCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACAAC
>GC:42030952:.:.:GTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGA:114.833:PASS:C1=12;C2=1378;EF=114.833;GC=0.309524;DU=0.375;VI=-1
GTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGA
>GC:42030952:.:.:TACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAA:270:PASS:C1=6;C2=1620;EF=270;GC=0.309524;DU=0.225;VI=-7.3
TACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTTGAA
>GC:42030952:.:.:TATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATG:115:PASS:C1=12;C2=1380;EF=115;GC=0.309524;DU=0.35;VI=-1.9
TATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATG
>GC:42030952:.:.:TCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCC:134.5:PASS:C1=10;C2=1345;EF=134.5;GC=0.309524;DU=0.375;VI=-3.1
TCAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCC
>GC:42030952:.:.:TCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACA:97.2667:PASS:C1=15;C2=1459;EF=97.2667;GC=0.309524;DU=0.3;VI=-7.3
TCCCTAAGCATTTGTTTCAGGGTTATTTGAATATCTATAACA
>GC:42030952:.:.:TGATACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTT:167.5:PASS:C1=10;C2=1675;EF=167.5;GC=0.309524;DU=0.175;VI=-7.3
TGATACCAATAAAATCCCTAAGCATTTGTTTCAGGGTTATTT
>GC:42033333:.:.:ATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGC:278:PASS:C1=5;C2=1390;EF=278;GC=0.333333;DU=0.275;VI=-2.4
ATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGC
>GC:42033333:.:.:CAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCG:109.083:PASS:C1=12;C2=1309;EF=109.083;GC=0.333333;DU=0.375;VI=-4.3
CAGGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCG
>GC:42033333:.:.:GGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTG:96.7857:PASS:C1=14;C2=1355;EF=96.7857;GC=0.333333;DU=0.375;VI=-4.3
GGGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTG
>GC:42033333:.:.:GGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGG:96.6429:PASS:C1=14;C2=1353;EF=96.6429;GC=0.333333;DU=0.375;VI=-4.2
GGTTATTTGAATATCTATAACAACTATTTTAAAGCGCCGTGG
>GC:42035714:.:.:TTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCT:130.182:PASS:C1=11;C2=1432;EF=130.182;GC=0.357143;DU=0.225;VI=-2.4
TTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCT
>GC:42035714:.:.:TTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCC:234.667:PASS:C1=6;C2=1408;EF=234.667;GC=0.357143;DU=0.25;VI=-2.4
TTTGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCC
>GC:42038095:.:.:AATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGAC:210.571:PASS:C1=7;C2=1474;EF=210.571;GC=0.380952;DU=0.225;VI=-2.4
AATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGAC
>GC:42038095:.:.:GAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGA:292.2:PASS:C1=5;C2=1461;EF=292.2;GC=0.380952;DU=0.225;VI=-2.4
GAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGA
>GC:42038095:.:.:TGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTG:179.75:PASS:C1=8;C2=1438;EF=179.75;GC=0.380952;DU=0.225;VI=-2.4
TGAATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTG
>GC:42040476:.:.:ATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACC:63.9583:PASS:C1=24;C2=1535;EF=63.9583;GC=0.404762;DU=0.225;VI=-2.4
ATATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACC
>GC:42042857:.:.:ATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGT:265:PASS:C1=6;C2=1590;EF=265;GC=0.428571;DU=0.175;VI=-1.4
ATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGT
>GC:42042857:.:.:TATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCG:98.0625:PASS:C1=16;C2=1569;EF=98.0625;GC=0.428571;DU=0.2;VI=-1.7
TATCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCG
>GC:42042857:.:.:TCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTA:265.167:PASS:C1=6;C2=1591;EF=265.167;GC=0.428571;DU=0.2;VI=-0.9
TCTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTA
>GC:42045238:.:.:CTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTAC:114.5:PASS:C1=14;C2=1603;EF=114.5;GC=0.452381;DU=0.2;VI=-0.9
CTATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTAC
>GC:42045238:.:.:TATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTACC:124.231:PASS:C1=13;C2=1615;EF=124.231;GC=0.452381;DU=0.175;VI=-1
TATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTACC
>GC:42047619:.:.:ATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTACCG:77:PASS:C1=21;C2=1617;EF=77;GC=0.47619;DU=0.175;VI=-2.2
ATAACAACTATTTTAAAGCGCCGTGGATGCCTGACCGTACCG
```

**Comment by torognes** [comment]
Hi @avilella,

I think this problem was fixed in version 1.0.13. Please the latest version, version 1.1.1.

- Torbjørn

**Comment by avilella** [comment]
Cool, I will try again. Sometimes the error is a strace like this:

======= Memory map: ========
00400000-00445000 r-xp 00000000 08:42 15341323                           /opt/foo/bar/thirdparty/vsearch-1.0.10/bin/vsearch-1.0.10-linux-x86_64
00645000-00649000 rw-p 00045000 08:42 15341323                           /opt/foo/bar/thirdparty/vsearch-1.0.10/bin/vsearch-1.0.10-linux-x86_64
00649000-0074b000 rw-p 00000000 00:00 0 
00b26000-01093000 rw-p 00000000 00:00 0                                  [heap]
330c000000-330c020000 r-xp 00000000 08:42 14155826                       /lib64/ld-2.12.so
330c21f000-330c220000 r--p 0001f000 08:42 14155826                       /lib64/ld-2.12.so
330c220000-330c221000 rw-p 00020000 08:42 14155826                       /lib64/ld-2.12.so
330c221000-330c222000 rw-p 00000000 00:00 0 
330c400000-330c58a000 r-xp 00000000 08:42 14155868                       /lib64/libc-2.12.so
330c58a000-330c78a000 ---p 0018a000 08:42 14155868                       /lib64/libc-2.12.so
330c78a000-330c78e000 r--p 0018a000 08:42 14155868                       /lib64/libc-2.12.so
330c78e000-330c78f000 rw-p 0018e000 08:42 14155868                       /lib64/libc-2.12.so
330c78f000-330c794000 rw-p 00000000 00:00 0 
330c800000-330c817000 r-xp 00000000 08:42 14155874                       /lib64/libpthread-2.12.so
330c817000-330ca17000 ---p 00017000 08:42 14155874                       /lib64/libpthread-2.12.so
330ca17000-330ca18000 r--p 00017000 08:42 14155874                       /lib64/libpthread-2.12.so
330ca18000-330ca19000 rw-p 00018000 08:42 14155874                       /lib64/libpthread-2.12.so
330ca19000-330ca1d000 rw-p 00000000 00:00 0 
330d000000-330d083000 r-xp 00000000 08:42 14155886                       /lib64/libm-2.12.so
330d083000-330d282000 ---p 00083000 08:42 14155886                       /lib64/libm-2.12.so
330d282000-330d283000 r--p 00082000 08:42 14155886                       /lib64/libm-2.12.so
330d283000-330d284000 rw-p 00083000 08:42 14155886                       /lib64/libm-2.12.so
3c0b400000-3c0b416000 r-xp 00000000 08:42 14155832                       /lib64/libgcc_s-4.4.7-20120601.so.1
3c0b416000-3c0b615000 ---p 00016000 08:42 14155832                       /lib64/libgcc_s-4.4.7-20120601.so.1
3c0b615000-3c0b616000 rw-p 00015000 08:42 14155832                       /lib64/libgcc_s-4.4.7-20120601.so.1
3c0b800000-3c0b8e8000 r-xp 00000000 08:42 18761972                       /usr/lib64/libstdc++.so.6.0.13
3c0b8e8000-3c0bae8000 ---p 000e8000 08:42 18761972                       /usr/lib64/libstdc++.so.6.0.13
3c0bae8000-3c0baef000 r--p 000e8000 08:42 18761972                       /usr/lib64/libstdc++.so.6.0.13
3c0baef000-3c0baf1000 rw-p 000ef000 08:42 18761972                       /usr/lib64/libstdc++.so.6.0.13
3c0baf1000-3c0bb06000 rw-p 00000000 00:00 0 
7f0264000000-7f0264021000 rw-p 00000000 00:00 0 
7f0264021000-7f0268000000 ---p 00000000 00:00 0 
7f026c000000-7f026c021000 rw-p 00000000 00:00 0 
7f026c021000-7f0270000000 ---p 00000000 00:00 0 
7f02733f8000-7f02733f9000 ---p 00000000 00:00 0 
7f02733f9000-7f0273df9000 rw-p 00000000 00:00 0 
7f02764fa000-7f02764fb000 ---p 00000000 00:00 0 
7f02764fb000-7f0276efb000 rw-p 00000000 00:00 0 
7f0276efb000-7f0276efc000 ---p 00000000 00:00 0 
7f0276efc000-7f02778fc000 rw-p 00000000 00:00 0 
7f0291d26000-7f0292830000 rw-p 00000000 00:00 0 
7f0292848000-7f029284b000 rw-p 00000000 00:00 0 
7fffd42d1000-7fffd43d7000 rw-p 00000000 00:00 0                          [stack]
7fffd4400000-7fffd4401000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
